### PR TITLE
Fix bug with mosviz load 1d spectra method

### PR DESCRIPTION
**Pull request by javerbukh**
### Description

This pull request is to address an issue with `mosviz.load_1d_spectra()` not linking data correctly.

Fixes #<Issue Number>

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
Should we add a regression test?

File: jdaviz/app.py [hunk 1]
```diff
@@ -422,7 +422,12 @@ def _link_new_data(self, reference_data=None, data_to_be_linked=None):
         # The glue-astronomy SpectralCoordinates currently seems incompatible with glue
         # WCSLink. This gets around it until there's an upstream fix.
         if isinstance(linked_data.coords, SpectralCoordinates):
-            wc_old = ref_data.world_component_ids[-1]
+            # Need the else case for mosviz.load_1d_spectra(), since
+            # the reference data will be 1D.
+            if len(ref_data.world_component_ids) > 1:
+                wc_old = ref_data.world_component_ids[-1]
+            else:
+                wc_old = ref_data.world_component_ids[0]
```
Comment (kecnry) [inline jdaviz/app.py:430]: the else case only covers `len(ref_data.world_component_ids) == 1`, right?  Why would `[0]` act differently than `[-1]` for a length of 1?

Comment (bmorris3) [reply jdaviz/app.py:430]: Wondering the same thing 🤔

**Comment by javerbukh** [comment]
Ok I think this was actually a problem with the environment (AKA user error). Closing now.

**Pull request closed without merging**